### PR TITLE
go 1.19 go@1.18 1.18.5

### PR DESCRIPTION
**Pull request by bayandin**
Created with `brew bump-formula-pr`.

`resource` blocks may require updates.

### CODE REVIEW

**Comment by carlocab** [comment]
We want `CI-build-dependents-from-source` for major version bumps, no?

**Comment by bayandin** [comment]
> We want `CI-build-dependents-from-source` for major version bumps, no?

Hmm, shouldn't the label be set automatically?

> Though we've got so many long-running PR jobs going I'm a bit tempted to skip it for now.

We could try, at least once.

**Comment by carlocab** [comment]
> Hmm, shouldn't the label be set automatically?

It should, but I think our triage workflow gets confused as soon as a PR touches more than one file.

**Comment by Bo98** [comment]
`go-critic` and `golangci-lint` need revision bumps.

**Comment by bayandin** [comment]
Here're the failures from 12-arm64; a bunch of them should be fixed by switching to go@1.18 (will do it as soon the CI run is done)

- `brew fetch --build-from-source --retry krakend`
```
  Warning: Formula reports different sha256: 82e602af5606ad5745e60bce0916ec18c598aab0d613fe7e5fd2ab55a71eaa53
```

- `brew fetch --build-from-source --retry whalebrew`
```
  Error: 0.4.0 tag should be 1722002db3f1618f7af9d400343bf502d03b508f
  but is actually bdf94887abf0397341c1d241974eea790626ae7c
```

- `brew install --build-from-source brev`
```
  imports inet.af/netstack/gohacks: build constraints exclude all Go files in /Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/inet.af/netstack@v0.0.0-20211120045802-8aa80cf23d3c/gohacks
```

- `brew install --build-from-source cloudflared`
```
  vendor/github.com/lucas-clemente/quic-go/internal/qtls/go119.go:6:13: cannot use "The version of quic-go you're using can't be built on Go 1.19 yet. For more details, please see https://github.com/lucas-clemente/quic-go/wiki/quic-go-and-Go-versions." (untyped string constant "The version of quic-go you're using can't be built on Go 1.19 yet. F...) as int value in variable declaration
```

- `brew install --build-from-source colima`
```
  	imports gvisor.dev/gvisor/pkg/gohacks: build constraints exclude all Go files in /Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/gvisor.dev/gvisor@v0.0.0-20210924063838-a1b00da7bd8d/pkg/gohacks
```

- `brew install --build-from-source doggo`
```
  /Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/lucas-clemente/quic-go@v0.26.0/internal/qtls/go119.go:6:13: cannot use "quic-go doesn't build on Go 1.19 yet." (untyped string constant) as int value in variable declaration
```

- `brew install --build-from-source flyctl`
```
  imports gvisor.dev/gvisor/pkg/gohacks: build constraints exclude all Go files in /Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/gvisor.dev/gvisor@v0.0.0-20211020211948-f76a604701b6/pkg/gohacks
```

- `brew install --build-from-source grafana`
```
  ➤ YN0000: │ grafana@workspace:. STDERR Error [ERR_LOADER_CHAIN_INCOMPLETE]: "file:///private/tmp/grafana-20220802-39279-liwzyz/grafana-9.0.2/.pnp.loader.mjs 'resolve'" did not call the next hook in its chain and did not explicitly signal a short circuit. If this is intentional, include `shortCircuit: true` in the hook's return.
  ➤ YN0000: │ grafana@workspace:. STDERR     at new NodeError (node:internal/errors:387:5)
  ➤ YN0000: │ grafana@workspace:. STDERR     at ESMLoader.resolve (node:internal/modules/esm/loader:860:13)
  ➤ YN0000: │ grafana@workspace:. STDERR     at async ESMLoader.getModuleJob (node:internal/modules/esm/loader:439:7)
  ➤ YN0000: │ grafana@workspace:. STDERR     at async Promise.all (index 0)
  ➤ YN0000: │ grafana@workspace:. STDERR     at async ESMLoader.import (node:internal/modules/esm/loader:541:24)
  ➤ YN0000: │ grafana@workspace:. STDERR     at async loadESM (node:internal/process/esm_loader:91:5)
  ➤ YN0000: │ grafana@workspace:. STDERR     at async handleMainPromise (node:internal/modules/run_main:65:12) {
  ➤ YN0000: │ grafana@workspace:. STDERR   code: 'ERR_LOADER_CHAIN_INCOMPLETE'
```

- `brew install --build-from-source ipfs`
```
  /Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/lucas-clemente/quic-go@v0.27.1/internal/qtls/go119.go:6:13: cannot use "The version of quic-go you're using can't be built on Go 1.19 yet. For more details, please see https://github.com/lucas-clemente/quic-go/wiki/quic-go-and-Go-versions." (untyped string constant "The version of quic-go you're using can't be built on Go 1.19 yet. F...) as int value in variable declaration
```

- `brew install --build-from-source istioctl`
```
  /Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/lucas-clemente/quic-go@v0.27.0/internal/qtls/go119.go:6:13: cannot use "The version of quic-go you're using can't be built on Go 1.19 yet. For more details, please see https://github.com/lucas-clemente/quic-go/wiki/quic-go-and-Go-versions." (untyped string constant "The version of quic-go you're using can't be built on Go 1.19 yet. F...) as int value in variable declaration
```

- `brew install --build-from-source k3d`
```
  ==> go build -mod vendor -ldflags=-s -w -X github.com/k3d-io/k3d/v5/version.Vers
  Error: Failure while executing; `/opt/homebrew/Cellar/k3d/5.4.4/bin/k3d completion bash` exited with 2. Here's the output:
```

- `brew install --build-from-source liqoctl`
```
  ==> go build -ldflags=-s -w -X github.com/liqotech/liqo/pkg/liqoctl/version.liqo
  Error: Failure while executing; `/opt/homebrew/Cellar/liqoctl/0.5.3/bin/liqoctl completion bash` exited with 2. Here's the output:
```

- `brew install --build-from-source octant`
```
  npm ERR! Invalid Version: ^3.0.8
  
  npm ERR! A complete log of this run can be found in:
```

- `brew install --build-from-source ooniprobe`
```
  /Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/ooni/psiphon/tunnel-core@v0.0.0-20220519122549-9c044eb6bd83/oovendor/quic-go/internal/qtls/go119.go:6:13: cannot use "quic-go doesn't build on Go 1.19 yet." (untyped string constant) as int value in variable declaration
```

- `brew install --build-from-source podman`
```
  imports gvisor.dev/gvisor/pkg/gohacks: build constraints exclude all Go files in /private/tmp/podman--gvproxy-20220803-21422-qmzyqt/gvisor-tap-vsock-0.3.0/vendor/gvisor.dev/gvisor/pkg/gohacks
```

- `brew install --build-from-source storj-uplink`
```
  /Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/lucas-clemente/quic-go@v0.25.0/internal/qtls/go119.go:6:13: cannot use "quic-go doesn't build on Go 1.19 yet." (untyped string constant) as int value in variable declaration
```

- `brew install --build-from-source sync_gateway`
```
  Error: Failure while executing; `python rewrite-manifest.sh --manifest-url file:///private/tmp/sync_gateway-20220802-6325-tr937d/manifest/default.xml --project-name sync_gateway --set-revision e54a62741bb28f3e54a6599c21c739df9a9dad76` exited with 127. Here's the output:
```

- `brew install --build-from-source syncthing`
```
  /Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/lucas-clemente/quic-go@v0.27.0/internal/qtls/go119.go:6:13: cannot use "The version of quic-go you're using can't be built on Go 1.19 yet. For more details, please see https://github.com/lucas-clemente/quic-go/wiki/quic-go-and-Go-versions." (untyped string constant "The version of quic-go you're using can't be built on Go 1.19 yet. F...) as int value in variable declaration
```

- `brew install --build-from-source traefik`
```
  /Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/lucas-clemente/quic-go@v0.27.0/internal/qtls/go119.go:6:13: cannot use "The version of quic-go you're using can't be built on Go 1.19 yet. For more details, please see https://github.com/lucas-clemente/quic-go/wiki/quic-go-and-Go-versions." (untyped string constant "The version of quic-go you're using can't be built on Go 1.19 yet. F...) as int value in variable declaration
```

- `brew install --build-from-source traefik@1`
```
  ERROR in ./src/app.sass
  Module build failed: Error: Cannot find module 'node-sass'
```

- `brew install --build-from-source vitess`
```
  ERROR: Go version reported: go version go1.19 darwin/arm64. Version 1.18.4+ required. See https://vitess.io/contributing/build-from-source for install instructions.
```

- `brew install --build-from-source xray`
```
  /Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/lucas-clemente/quic-go@v0.27.0/internal/qtls/go119.go:6:13: cannot use "The version of quic-go you're using can't be built on Go 1.19 yet. For more details, please see https://github.com/lucas-clemente/quic-go/wiki/quic-go-and-Go-versions." (untyped string constant "The version of quic-go you're using can't be built on Go 1.19 yet. F...) as int value in variable declaration
```

- `brew install --build-from-source yorkie`
```
  # github.com/yorkie-team/yorkie/pkg/document/json
  pkg/document/json/rga_tree_split.go:302:29: internal compiler error: method Len on *uint8 not found
```

- `brew test --retry --verbose gomplate`
```
  panic: Something in this program imports go4.org/unsafe/assume-no-moving-gc to declare that it assumes a non-moving garbage collector, but your version of go4.org/unsafe/assume-no-moving-gc hasn't been updated to assert that it's safe against the go1.19 runtime. If you want to risk it, run with environment variable ASSUME_NO_MOVING_GC_UNSAFE_RISK_IT_WITH=go1.19 set. Notably, if go1.19 adds a moving garbage collector, this program is unsafe to use.
```

- `brew test --retry --verbose grafana-agent`
```
  panic: Something in this program imports go4.org/unsafe/assume-no-moving-gc to declare that it assumes a non-moving garbage collector, but your version of go4.org/unsafe/assume-no-moving-gc hasn't been updated to assert that it's safe against the go1.19 runtime. If you want to risk it, run with environment variable ASSUME_NO_MOVING_GC_UNSAFE_RISK_IT_WITH=go1.19 set. Notably, if go1.19 adds a moving garbage collector, this program is unsafe to use.
```

- `brew test --retry --verbose logcli`
```
  panic: Something in this program imports go4.org/unsafe/assume-no-moving-gc to declare that it assumes a non-moving garbage collector, but your version of go4.org/unsafe/assume-no-moving-gc hasn't been updated to assert that it's safe against the go1.19 runtime. If you want to risk it, run with environment variable ASSUME_NO_MOVING_GC_UNSAFE_RISK_IT_WITH=go1.19 set. Notably, if go1.19 adds a moving garbage collector, this program is unsafe to use.
```

- `brew test --retry --verbose promtail`
```
  panic: Something in this program imports go4.org/unsafe/assume-no-moving-gc to declare that it assumes a non-moving garbage collector, but your version of go4.org/unsafe/assume-no-moving-gc hasn't been updated to assert that it's safe against the go1.19 runtime. If you want to risk it, run with environment variable ASSUME_NO_MOVING_GC_UNSAFE_RISK_IT_WITH=go1.19 set. Notably, if go1.19 adds a moving garbage collector, this program is unsafe to use.
```

**Comment by carlocab** [comment]
`octant` is pre-existing. #106816

Most of the rest (build constraints, garbage collector, Go version complaints) can probably be fixed by building with a `go@1.18`.

That still leaves a few failures, though.

**Comment by andig** [comment]
@carlocab are you planning to fix Go 1.18 for part of the packages? I could help tracking the bug ones.

**Comment by bayandin** [comment]
> a bunch of them should be fixed by switching to go@1.18 (will do it as soon the CI run is done)

**Comment by andig** [comment]
For https://github.com/go4org/unsafe-assume-no-moving-gc 1.19 is already marked safe, I'll open upstream issues.

**Comment by bayandin** [comment]
Formulae that require some attention (has unknown failure):

- `brew install --build-from-source grafana`
- `brew install --build-from-source k3d`
- `brew install --build-from-source liqoctl`
- `brew install --build-from-source sync_gateway`
- `brew install --build-from-source traefik@1`

**Comment by carlocab** [comment]
I think `grafana` is also pre-existing. The error looks similar to https://github.com/Homebrew/homebrew-core/pull/107028.

**Comment by andig** [comment]
`traefik@1` is not related to Go at all- no idea. Must be pre-existing?

**Comment by carlocab** [comment]
> `traefik@1` is not related to Go at all- no idea. Must be pre-existing?

Maybe. Do you see a similar error with
```
brew install -s traefik@1
```
?

**Comment by andig** [comment]
> > `traefik@1` is not related to Go at all- no idea. Must be pre-existing?
> 
> Maybe. Do you see a similar error with
> 
> ```
> brew install -s traefik@1
> ```
> 
> ?

Traefik has a different issue for me, but still broken:

```
==> Installing traefik@1
==> yarn upgrade
==> yarn install
==> yarn run build
Last 15 lines from /Users/andig/Library/Logs/Homebrew/traefik@1/03.yarn:
    at loadLoader (/private/tmp/traefikA1-20220803-18031-17oz0fj/traefik-1.7.34/webui/node_modules/loader-runner/lib/loadLoader.js:32:11)
    at iteratePitchingLoaders (/private/tmp/traefikA1-20220803-18031-17oz0fj/traefik-1.7.34/webui/node_modules/loader-runner/lib/LoaderRunner.js:169:2)
    at iteratePitchingLoaders (/private/tmp/traefikA1-20220803-18031-17oz0fj/traefik-1.7.34/webui/node_modules/loader-runner/lib/LoaderRunner.js:165:10)
    at /private/tmp/traefikA1-20220803-18031-17oz0fj/traefik-1.7.34/webui/node_modules/loader-runner/lib/LoaderRunner.js:176:18
    at loadLoader (/private/tmp/traefikA1-20220803-18031-17oz0fj/traefik-1.7.34/webui/node_modules/loader-runner/lib/loadLoader.js:47:3)
    at iteratePitchingLoaders (/private/tmp/traefikA1-20220803-18031-17oz0fj/traefik-1.7.34/webui/node_modules/loader-runner/lib/LoaderRunner.js:169:2)
    at iteratePitchingLoaders (/private/tmp/traefikA1-20220803-18031-17oz0fj/traefik-1.7.34/webui/node_modules/loader-runner/lib/LoaderRunner.js:165:10)
    at /private/tmp/traefikA1-20220803-18031-17oz0fj/traefik-1.7.34/webui/node_modules/loader-runner/lib/LoaderRunner.js:176:18
    at loadLoader (/private/tmp/traefikA1-20220803-18031-17oz0fj/traefik-1.7.34/webui/node_modules/loader-runner/lib/loadLoader.js:47:3)
    at iteratePitchingLoaders (/private/tmp/traefikA1-20220803-18031-17oz0fj/traefik-1.7.34/webui/node_modules/loader-runner/lib/LoaderRunner.js:169:2)
    at runLoaders (/private/tmp/traefikA1-20220803-18031-17oz0fj/traefik-1.7.34/webui/node_modules/loader-runner/lib/LoaderRunner.js:365:2)
    at NormalModule.doBuild (/private/tmp/traefikA1-20220803-18031-17oz0fj/traefik-1.7.34/webui/node_modules/webpack/lib/NormalModule.js:182:3)
    at NormalModule.build (/private/tmp/traefikA1-20220803-18031-17oz0fj/traefik-1.7.34/webui/node_modules/webpack/lib/NormalModule.js:275:15)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by andig** [comment]
- k3d currently works
- liqoctl currently works
- sync_gateway currently **broken**

**Comment by andig** [comment]
Not sure about liqo: it does build with 1.19, dunno why CI fails:

Go 1.18:

```
brew reinstall --build-from-source liqoctl
==> Downloading https://github.com/liqotech/liqo/archive/refs/tags/v0.5.3.tar.gz
Already downloaded: /Users/andig/Library/Caches/Homebrew/downloads/cb26d7a9b0cbb07e56b0a292ca99a59c205a69c9f5a9909ab9d006bbfde2d1c6--liqo-0.5.3.tar.gz
==> Reinstalling liqoctl
==> go build -ldflags=-s -w -X github.com/liqotech/liqo/pkg/liqoctl/version.liqoctlVersion=v0.5.3 ./cmd/liqoctl
==> Caveats
zsh completions have been installed to:
  /opt/homebrew/share/zsh/site-functions
==> Summary
🍺  /opt/homebrew/Cellar/liqoctl/0.5.3: 8 files, 58.7MB, built in 6 seconds
==> Running `brew cleanup liqoctl`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
```

Go 1.19:

    go1.19 build -ldflags="-s -w -X github.com/liqotech/liqo/pkg/liqoctl/version.liqoctlVersion=v0.5.3" ./cmd/liqoctl

**Comment by bayandin** [comment]
> Not sure about `liqo`: it does build with 1.19, dunno why CI fails:
> 
> ...
>
> Go 1.19:
> 
> ```
> go1.19 build -ldflags="-s -w -X github.com/liqotech/liqo/pkg/liqoctl/version.liqoctlVersion=v0.5.3" ./cmd/liqoctl
> ```

It fails to generate completions. Could you try to run `liqoctl completion bash` and see if it works (for go1.19 built liqoctl)

**Comment by andig** [comment]
Same root cause (only this time with completions, not compile), will open issue.

**Comment by andig** [comment]
The fixes for go4.org are underway but likely releases will take their time (https://github.com/grafana/agent/issues/1983#issuecomment-1204123898). Fix to 1.18 in the mean time or wait?

**Comment by bayandin** [comment]
I've pushed everything I have.
Meanwhile, while we're waiting for a free `CI-long-timeout` slot, feel free to check failed formulae on other systems (I've checked only 12-arm64 runner). Here's a link to the previous run: https://github.com/Homebrew/homebrew-core/actions/runs/2785592300

**Comment by dirien** [comment]
Awesome, can't wait to get go 1.19 via homebrew! Thanks @bayandin!

File: Formula/gomplate.rb [hunk 1]
```diff
@@ -15,7 +15,9 @@ class Gomplate < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "3703f68a2d7c4598da6a0ec112ed38bb79a61073ffc161728721dd2e61895e9b"
   end
 
-  depends_on "go" => :build
+  # Required latest https://pkg.go.dev/go4.org/unsafe/assume-no-moving-gc
+  # Try to switch to the latest go on the next release
+  depends_on "go@1.18" => :build
```
Comment (hairyhenderson) [inline Formula/gomplate.rb:20]: FYI once https://github.com/hairyhenderson/gomplate/pull/1469 merges I'll release gomplate 3.11.2, which should make this change unnecessary

**Comment by bayandin** [comment]
So far, after 12-arm64, 11-arm64, linux:

**Unknown:**
- `brew test --retry --verbose func-e` (on 11-arm64 only)
```
 ==> Testing func-e
  ==> /opt/homebrew/Cellar/func-e/1.1.3/bin/func-e run --version
  dyld: Symbol not found: __ZNKSt3__115basic_stringbufIcNS_11char_traitsIcEENS_9allocatorIcEEE3strEv
    Referenced from: /private/tmp/func-e-test-20220805-14034-7m57hu/.func-e/versions/1.22.2/bin/envoy (which was built for Mac OS X 12.3)
    Expected in: /usr/lib/libc++.1.dylib
```
- `brew install --build-from-source sync_gateway` (on 12-arm64 only)
```
 Error: Failure while executing; `python rewrite-manifest.sh --manifest-url file:///private/tmp/sync_gateway-20220805-623-avs958/manifest/default.xml --project-name sync_gateway --set-revision e54a62741bb28f3e54a6599c21c739df9a9dad76` exited with 127. Here's the output:
```

**Switch to 1.18:**
- `brew install --build-from-source yorkie`
- `brew test --retry --verbose loki`

**Pre-existing:**
- ` brew fetch --build-from-source --retry krakend # https://github.com/krakendio/krakend-ce/issues/541`
- `brew fetch --build-from-source --retry whalebrew # https://github.com/whalebrew/whalebrew/issues/192`
- `brew install --build-from-source octant # preexisting https://github.com/Homebrew/homebrew-core/pull/106816`
- `brew install --verbose --build-bottle istioctl # preexisting https://github.com/Homebrew/homebrew-core/pull/107108`

**Comment by carlocab** [comment]
I suggest moving the `go@1.18` addition along with its dependents (the ones that build, anyway) into a separate PR to help simplify this one.

We also don't need new bottles for the ones that just have a build-only dependency on `go@1.18` anyway.

**Comment by bayandin** [comment]
> I suggest moving the `go@1.18` addition along with its dependents (the ones that build, anyway) into a separate PR to help simplify this one.
> 
> We also don't need new bottles for the ones that just have a build-only dependency on `go@1.18` anyway.

Ok, let's try!

**Review by carlocab — DISMISSED** [review]

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**